x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -288,39 +297,86 @@ protected LogicalPlan rule(UnaryPlan plan) {
             if (plan instanceof Project project) {
                 if (child instanceof Project p) {
                     // eliminate lower project but first replace the aliases in the upper one
-                    return p.withProjections(combineProjections(project.projections(), p.projections()));
-                } else if (child instanceof Aggregate a) {
+                    project = p.withProjections(combineProjections(project.projections(), p.projections()));
+                    child = project.child();
+                    plan = project;
+                    // don't return the plan since the grandchild (now child) might be an aggregate that could not be folded on the way up
+                    // e.g. stats c = count(x) | project c, c as x | project x
+                    // try to apply the rule again opportunistically as another node might be pushed in (a limit might be pushed in)
+                }
+                // check if the projection eliminates certain aggregates
+                // but be mindful of aliases to existing aggregates that we don't want to duplicate to avoid redundant work
+                if (child instanceof Aggregate a) {
                     var aggs = a.aggregates();
-                    var newAggs = combineProjections(project.projections(), aggs);
-                    var newGroups = replacePrunedAliasesUsedInGroupBy(a.groupings(), aggs, newAggs);
-                    return new Aggregate(a.source(), a.child(), newGroups, newAggs);
+                    var tuple = projectAggregations(project.projections(), aggs);
+                    // project can be fully removed
+                    if (tuple.v1().isEmpty()) {
+                        var newAggs = tuple.v2();
+                        var newGroups = replacePrunedAliasesUsedInGroupBy(a.groupings(), aggs, newAggs);
+                        plan = new Aggregate(a.source(), a.child(), newGroups, newAggs);
+                    }
```

Comment:
The gist of this change takes care of the scenario creates by  `ReplaceStatsAggExpressionWithEval`:
```
stats x = sum(), y = count()
project x, x as a, y
```

The combine rule previously would combine project into stats which would duplicate the count:
```
stats x = sum(), a = sum(), y = count()
```
It removes the project (which is cheap) but duplicates the sum (which is expensive and the reason we didn't want to duplicate it in the first place).

The rule thus tracks is there's any new alias - however it keeps on removing unused aggregations and in case of  basic aliasing project, removes the project.
So the following
```
stats x = sum(), y = count()
project x as a
```

becomes
```
stats a = sum()
```